front/pages/w/[wId]/builder/labs/transcripts/index.tsx

```diff
@@ -0,0 +1,335 @@
+import AppLayout from "@app/components/sparkle/AppLayout";
+import { subNavigationBuild } from "@app/components/sparkle/navigation";
+import { SendNotificationsContext } from "@app/components/sparkle/Notification";
+import apiConfig from "@app/lib/api/config";
+import { buildConnectionId } from "@app/lib/connector_connection_id";
+import { withDefaultUserAuthRequirements } from "@app/lib/iam/session";
+import config from "@app/lib/labs/config";
+import {
+  useAgentConfigurations,
+  useLabsTranscriptsConfiguration,
+} from "@app/lib/swr";
+
+const provider = "google_drive";
+
+export const getServerSideProps = withDefaultUserAuthRequirements<{
+  owner: WorkspaceType;
+  subscription: SubscriptionType;
+  gaTrackingId: string;
+  nangoDriveConnectorId: string;
+  nangoPublicKey: string;
+}>(async (_context, auth) => {
+  const owner = auth.workspace();
+  const subscription = auth.subscription();
+
+  if (!owner || !subscription) {
+    return {
+      notFound: true,
+    };
+  }
+
+  return {
+    props: {
+      owner,
+      subscription,
+      gaTrackingId: apiConfig.getGaTrackingId(),
+      nangoDriveConnectorId: config.getNangoGoogleDriveConnectorId(),
+      nangoPublicKey: config.getNangoPublicKey(),
+    },
+  };
+});
+
+export default function LabsTranscriptsIndex({
+  owner,
+  subscription,
+  gaTrackingId,
+  nangoDriveConnectorId,
+  nangoPublicKey,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const sendNotification = useContext(SendNotificationsContext);
+
+  const { agentConfigurations } = useAgentConfigurations({
+    workspaceId: owner.sId,
+    agentsGetView: "list",
+    sort: "priority",
+  });
+
+  const { labsConfiguration, islabsConfigurationLoading } =
+    useLabsTranscriptsConfiguration({
+      workspaceId: owner.sId,
+      provider,
+    });
+
+  const [transcriptsConfigurationState, setTranscriptsConfigurationState] =
+    useState<{
+      isGDriveConnected: boolean;
+      assistantSelected: LightAgentConfigurationType | null;
+      emailToNotify: string;
+      isActive: boolean;
+    }>({
+      isGDriveConnected: false,
+      assistantSelected: null as LightAgentConfigurationType | null,
+      emailToNotify: "",
+      isActive: false,
+    });
+
+  useEffect(() => {
+    setTranscriptsConfigurationState({
+      isGDriveConnected:
+        (labsConfiguration && labsConfiguration.id > 0) || false,
+      assistantSelected:
+        agentConfigurations.find(
+          (a) => a.sId === labsConfiguration?.agentConfigurationId
+        ) || null,
+      emailToNotify: labsConfiguration?.emailToNotify || "",
+      isActive: labsConfiguration?.isActive || false,
+    });
+  }, [labsConfiguration, agentConfigurations]);
+
+  if (islabsConfigurationLoading) {
+    return <Spinner2 />;
+  }
+
+  const agents = agentConfigurations.filter((a) => a.status === "active");
+
+  const makePatchRequest = async (data: any, successMessage: string) => {
```

Comment:
You should export a type from `PatchLabsTranscriptsConfigurationBodySchema` and use `Partial<>` here instead of any.